care/facility/migrations/0429_double_pain_scale.py

```diff
@@ -0,0 +1,92 @@
+# Generated by Django 4.2.10 on 2024-03-12 17:48
+
+from math import ceil
+
+from django.core.paginator import Paginator
+from django.db import migrations, models
+
+import care.utils.models.validators
+
+
+def double_pain_scale(apps, schema_editor):
+    DailyRound = apps.get_model("facility", "DailyRound")
+
+    page = Paginator(
+        DailyRound.objects.only("id", "pain_scale_enhanced")
+        .exclude(pain_scale_enhanced__exact=[])
+        .order_by("id"),
+        2000,
+    )
+
+    for page_num in page.page_range:
+        records_to_update = []
+        for daily_round in page.page(page_num):
+            for obj in daily_round.pain_scale_enhanced:
+                try:
+                    obj["scale"] *= 2
+                except KeyError:
+                    pass
+            records_to_update.append(daily_round)
+        DailyRound.objects.bulk_update(records_to_update, ["pain_scale_enhanced"])
+
+
+def halve_pain_scale(apps, schema_editor):
+    DailyRound = apps.get_model("facility", "DailyRound")
+    page = Paginator(
+        DailyRound.objects.only("id", "pain_scale_enhanced")
+        .exclude(pain_scale_enhanced__exact=[])
+        .order_by("id"),
+        2000,
+    )
+
+    for page_num in page.page_range:
+        records_to_update = []
+        for daily_round in page.page(page_num):
+            for obj in daily_round.pain_scale_enhanced:
+                try:
+                    obj["scale"] = ceil(obj["scale"] / 2)
+                except KeyError:
+                    pass
+            records_to_update.append(daily_round)
+        DailyRound.objects.bulk_update(records_to_update, ["pain_scale_enhanced"])
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("facility", "0428_alter_patientmetainfo_occupation"),
+    ]
+
+    operations = [
+        migrations.RunPython(double_pain_scale, reverse_code=halve_pain_scale),
+        migrations.AlterField(
```

Comment:
shouldnt the change come first ? because we are writing values higher than 10 before the change is made.

Reply:
That is a model level validation, it won't affect the migrations